store/src/java/com/zimbra/cs/account/ZimbraAuthToken.java

```diff
@@ -499,25 +500,28 @@ public void deRegister() throws AuthTokenException {
 
     @Override
     public String getEncoded() throws AuthTokenException {
-        if (encoded == null) {
-            encoded = generateEncodedData();
-        }
-
         if (isJWT()) {
             try {
                 Account acct = Provisioning.getInstance().get(AccountBy.id, accountId);
                 ZimbraLog.account.debug("auth: generating jwt token for account id: %s", accountId);
-                AuthTokenKey authkey = getCurrentKey();
-                Key key = new SecretKeySpec(authkey.getKey(), SignatureAlgorithm.HS512.getJcaName());
+                RandomStringGenerator generator = new RandomStringGenerator.Builder().withinRange('a', 'z').build();
+                salt = generator.generate(20);
```

Comment:
Do you want to seed the RandomGenerator with something like username or accountId to ensure that the salt is unique. Also can we extract 20 into a constant easier to read the code.